### DISCUSSION TITLE
Remove Signals and Messages from Loaded Database

### DISCUSSION THREAD

**Discussion by johannes39** [cantools/cantools#419] [Q&A]
I would like to delete signals and messages from a database that I've loaded. Is this currently supported?

**Comment by andlaus** [comment]
for removing the message named `rouge_message` from a database `db`, use 
```python
msg = db.get_message_by_name("rouge_message")
i = db.message.index(msg)
del db.messages[i]
```

removing signals from messages is analogous except that `message.signals` must be used and after you're done modifying the message, you should call `message.refresh()` if you want to use the modified message object to encode or decode frames...
